### PR TITLE
Bug #17 fix: Add configuration to format date to display

### PR DESCRIPTION
**Pull request by paritshivani**

### CODE REVIEW

File: src/components/com_tc/languages/administrator/en-GB/en-GB.com_tc.ini [hunk 1]
```diff
@@ -101,11 +101,18 @@ COM_TC_CONTENTS_LATEST="Latest"
...
+COM_TC_CONTENTS_CLIENT_FILTER_DESC="T&C code"
 COM_TC_CONTENTS_VERSION_ASC="Version ascending"
 COM_TC_CONTENTS_VERSION_DES="Version descending"
-COM_TC_CONTENTS_CLIENT_ASC="Client ascending"
-COM_TC_CONTENTS_CLIENT_DES="Client descending"
+COM_TC_CONTENTS_CLIENT_ASC="T&C code ascending"
+COM_TC_CONTENTS_CLIENT_DES="T&C code descending"
 COM_TC_CONTENTS_STARTDATE_ASC="Startdate ascending"
 COM_TC_CONTENTS_STARTDATE_DES="Startdate descending"
+COM_TC_CONFIRM_DELETE_USERTCS="Are you sure you want to delete the user(s)?"
```
Comment (snehal0904) [inline src/components/com_tc/languages/administrator/en-GB/en-GB.com_tc.ini]: @paritshivani Are you sure you want to delete the user consents

File: src/components/com_tc/administrator/assets/elements/dateformat.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ * @package     Tc
+ * @subpackage  com_tc
+ *
+ * @author      Techjoomla <extensions@techjoomla.com>
+ * @copyright   Copyright (C) 2009 - 2020 Techjoomla. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die('Restricted access');
+jimport('joomla.form.formfield');
```
Comment (manojLondhe) [inline src/components/com_tc/administrator/assets/elements/dateformat.php]: use new namespaces classes

File: src/components/com_tc/administrator/assets/elements/dateformat.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	 *
+	 * @return  array  date format list
+	 *
+	 * @since   1.0
+	 */
+	public function fetchElement ($name, $value, &$node, $control_name)
+	{
+		$sqlGmtTimestamp = "2012-01-01 20:00:00";
+
+		$dateFormat = array("Y-m-d H:i:s",
```
Comment (manojLondhe) [inline src/components/com_tc/administrator/assets/elements/dateformat.php]: format correctly

File: src/components/com_tc/administrator/assets/elements/dateformat.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+							"D, M d h:i A", "F j, Y, g:i a", "m.d.y", "j, n, Y", "h-i-s, j-m-y", "H:i:s"
+							);
+
+		foreach ($dateFormat as $date)
+		{
+			$options[] = JHTML::_('select.option', $date, JHtml::date(
+								$sqlGmtTimestamp, $date, true
+								)
+							);
+		}
```
Comment (manojLondhe) [inline src/components/com_tc/administrator/assets/elements/dateformat.php:71]: format correctly

File: src/components/com_tc/site/views/content/tmpl/default.php [hunk 5]
```diff
@@ -27,12 +30,13 @@ function validateform(form)
...
 		<?php if (!empty($this->tc_id))
-		{?>
+		{
+			?>
 			<div class="well well-condensed">
 			<div class="">
 				<h1><?php echo $this->termsandconditions->title; ?></h1>
 				<strong><?php echo JText::_("COM_TC_LATEST_TERMSANDCONDITIONS_VERSION") . $this->termsandconditions->version; ?>&nbsp;&nbsp;
-				<?php echo JText::_("COM_TC_LATEST_TERMSANDCONDITIONS_UPDATED_DATE") . $this->termsandconditions->modified_on; ?></strong>
+				<?php echo JText::_("COM_TC_LATEST_TERMSANDCONDITIONS_UPDATED_DATE") . HTMLHelper::_('date', $this->termsandconditions->modified_on, $this->dateFormat, true); ?><strong>
```
Comment (manojLondhe) [inline src/components/com_tc/site/views/content/tmpl/default.php]: `<strong> `or` </strong>`

File: src/components/com_tc/languages/administrator/en-GB/en-GB.com_tc.ini [hunk 6]
```diff
@@ -101,11 +101,18 @@ COM_TC_CONTENTS_LATEST="Latest"
...
+COM_TC_CONTENTS_CLIENT_DES="T&C code descending"
 COM_TC_CONTENTS_STARTDATE_ASC="Startdate ascending"
 COM_TC_CONTENTS_STARTDATE_DES="Startdate descending"
+COM_TC_CONFIRM_DELETE_USERTCS="Are you sure you want to delete the user consents?"
+COM_TC_BASIC_CONFIGUE_SET="Basic configuration"
+COM_TC_DATE_FORMAT_TO_SHOW="Date format to show"
+COM_TC_DATE_FORMAT_TO_SHOW_DESC="You can set the format to display the date on the site"
+COM_TC_CUSTOM_DATE_FORMAT="Custom date format"
+COM_TC_CUSTOM_DATE_FORMAT_DESC="You can customize the date format here"
+COM_TC_DATE_FORMAT_CUSTOME="custom"
```
Comment (manojLondhe) [inline src/components/com_tc/languages/administrator/en-GB/en-GB.com_tc.ini]: COM_TC_DATE_FORMAT_CUSTOM

File: src/components/com_tc/administrator/views/usertcs/tmpl/default.php [hunk 7]
```diff
@@ -93,7 +111,10 @@
 				</tfoot>
 				<tbody>
 				<?php
-				foreach ($this->items as $i => $item) : ?>
+				foreach ($this->items as $i => $item) :
+
+						$acceptedDate = HTMLHelper::_('date', $item->accepted_date, $this->dateFormat, true);
```
Comment (manojLondhe) [inline src/components/com_tc/administrator/views/usertcs/tmpl/default.php]: do one tab less indent

File: src/components/com_tc/administrator/views/usertcs/tmpl/default.php [hunk 8]
```diff
@@ -19,6 +20,23 @@
 $listDirn  = $this->state->get('list.direction');
 ?>
 <script type="text/javascript">
+
+Joomla.submitbutton = function(action)
+	{
+		 if (action=='usertcs.delete')
+		{
+			var confirmation=confirm("<?php echo JText::_('COM_TC_CONFIRM_DELETE_USERTCS');?>");
+			if (confirmation==true)
```
Comment (manojLondhe) [inline src/components/com_tc/administrator/views/usertcs/tmpl/default.php]: for Boolean values use `===`

File: src/components/com_tc/administrator/assets/elements/dateformat.php [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+	 */
+	protected $type = 'dateformat';
+
+	/**
+	 * Fiedd to decide if options are being loaded externally and from xml
+	 *
+	 * @var   integer
+	 * @since 2.2
+	 */
+	protected $loadExternally = 0;
```
Comment (manojLondhe) [inline src/components/com_tc/administrator/assets/elements/dateformat.php]: where is this var used?

File: src/components/com_tc/administrator/assets/elements/dateformat.php [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+use Joomla\CMS\Form\FormField;
+use Joomla\CMS\HTML\HTMLHelper;
+/**
+ * JFormFieldDateformat class
+ *
+ * @package     Tc
+ * @subpackage  component
+ * @since       1.0
+ */
+class JFormFieldDateformat extends JFormField
```
Comment (snehal0904) [inline src/components/com_tc/administrator/assets/elements/dateformat.php]: @paritshivani extend to FormField as you have included its namespace

**Pull request merged**